rust/parquet/Cargo.toml

```diff
@@ -43,6 +43,8 @@ chrono = "0.4"
 num-bigint = "0.3"
 arrow = { path = "../arrow", version = "4.0.0-SNAPSHOT", optional = true }
 base64 = { version = "0.12", optional = true }
+clap = "2.33.3"
```

Comment:
would it be possible to make the `clap` dependency optional as well? Or maybe we can move the command line tools into their own feature?  The hope for parquet is that it becomes a widely used library at the lower levels of a software stack so I think keeping dependencies minimal is a good goal

@nevi-me  what do you think?

Reply:
Thanks @alamb ! Just to confirm: Having one feature for command line tools that has both `clap` + `serde_json` as dependencies and when used without that feature, no binaries are produced. Is my understanding correct?

Reply:
That is what I would suggest -- do you have any thoughts @nevi-me  or @sunchao ?

Reply:
Thanks @alamb - I've added a new feature called `cli` that encapsulates the binaries and has `serde_json` and `clap` as dependencies.